classes/class-wc-connect-nux.php

```diff
@@ -448,6 +458,18 @@ public function set_up_nux_notices() {
 					add_action( 'admin_notices', array( $this, 'show_banner_after_connection' ) );
 					break;
 			}
+
+			add_action( 'wp_ajax_wc_connect_dismiss_banner', array( $this, 'ajax_dismiss_banner' ) );
+		}
+
+		public function ajax_dismiss_banner() {
+			if ( empty( $_POST['dismiss_option'] ) ) {
+				return;
+			}
+
+			check_ajax_referer( 'wc_connect_dismiss_banner', 'nonce' );
+			update_option( 'wc_connect_dismiss_banner_' . $_POST['dismiss_option'], 'yes' );
```

Comment:
> I thought it would be helpful to have a generalized way to make any banner dismissible

Good thinking. How about in `show_nux_banner()` we check for the option that dismisses the banner if it has `dismiss_option` defined. Let's put those options in the compact options using `WC_Connect_Options`.

Reply:
Checking for the option in `show_nux_banner` now, but I've since realized that there already appears to be code for this, e.g. https://github.com/Automattic/woocommerce-services/blob/960c12b97878a3344f43e38a293c61417c838ca6/classes/class-wc-connect-nux.php#L40-L60
...though it's apparently only used for service settings notices. Later, I can look into using it rather than duplicating.

Reply:
Updated to use those existing functions for setting and retrieving dismissal state.